src/main/scala/io/citrine/lolo/trees/splits/BoltzmannSplitter.scala

```diff
@@ -0,0 +1,226 @@
+package io.citrine.lolo.trees.splits
+
+import io.citrine.lolo.trees.impurity.VarianceCalculator
+import io.citrine.lolo.trees.splits.BoltzmannSplitter.SplitterResult
+
+import scala.collection.mutable
+import scala.util.Random
+
+/**
+  * Find a split for a regression problem
+  *
+  * The splits are picked with a probability that is related to the reduction in variance:
+  * P(split) ~ exp[ - {reduction in variance} / ({temperature} * {total variance}) ]
+  * recalling that the "variance" here is weighted by the sample size (so its really the sum of the square difference
+  * from the mean).  This is akin to kinetic monte carlo and simulated annealing techniques.
+  *
+  * Created by maxhutch on 11/29/16.
+  *
+  * @param temperature used to control how sensitive the probability of a split is to its change in variance
+  */
+case class BoltzmannSplitter(temperature: Double) extends Splitter[Double] {
```

Comment:
Should there be a hard requirement that `temperature > 0`?

Reply:
I think that `temperature = 0` (as is allowed here) will cause problems. Because Scala deal with infinities well it won't throw any errors right away, but you'll get `score = 0` for all the splits and `draw = 0`, so you'll end up triggering the error on lines 89/164/237